### PR TITLE
updating vignette to include what a model_out_tbl is

### PR DESCRIPTION
**Pull request by nickreich**
Also, a few misc updates to package documentation that happened when I re-ran roxygen.

### CODE REVIEW

**Comment by nickreich** [comment]
This would close #113 at least for now.

File: vignettes/articles/connect_hub.Rmd [hunk 1]
```diff
@@ -9,7 +9,21 @@ knitr::opts_chunk$set(
 )
 ```
 
-An important function of `hubUtils` is allowing for the connection to data in the `model-output` directory to facilitate interrogation, filtering and extraction of Hub data.
+An important function of `hubUtils` is allowing for the connection to data in the `model-output` directory to facilitate extraction, filtering, querying, exploring, and analyzing of Hub data.
+
+# Structure of hubverse datasets
+
+All data returned from connecting to and querying hubs can be read or validated as a `model_out_tbl` which is an S3 class defined by the `hubUtils` package. A `model_out_tbl` is a long-form `data.frame` designed to conform to the [hubverse data specifications for model output data](https://hubdocs.readthedocs.io/en/latest/format/model-output.html). In short, the columns of a valid `model_out_tbl` containing model output data from a hub are:
```
Comment (annakrystalli) [inline vignettes/articles/connect_hub.Rmd:16]: We actually output a `tibble`. Not sure it's that important but given you explicitly mention `data.frame` thought I'd note that.

File: vignettes/articles/connect_hub.Rmd [hunk 2]
```diff
@@ -9,7 +9,21 @@ knitr::opts_chunk$set(
...
+
+ - `model_id`: this is the unique character identifier of a model.
```
Comment (annakrystalli) [inline vignettes/articles/connect_hub.Rmd:18]: We could mention here that `model_id` == `<team_abbr>_<model_abbr>`?

File: vignettes/articles/connect_hub.Rmd [hunk 3]
```diff
@@ -9,7 +9,21 @@ knitr::opts_chunk$set(
...
+ - `output_type`: a character variable that defines the type of representation of model output that is in a given row.
+ - `output_type_id`: a variable that specifies some additional identifying information specific to the output type in a given row, e.g., a numeric quantile level, a string giving the name of a possible category for a discrete outcome, or an index of a sample.
+ - `value`: a numeric variable that provides the information about the model's prediction.
+ - `...` : other columns will be present depending on modeling tasks defined by the individual modeling hub. These columns are referred to in hubverse lingo as the `task-ID` variables.
```
Comment (annakrystalli) [inline vignettes/articles/connect_hub.Rmd:22]:
```suggestion
 - `...` : other columns will be present depending on modeling tasks defined by the individual modeling hub. These columns are referred to in hubverse terminology as the `task-ID` variables.
```

File: vignettes/articles/connect_hub.Rmd [hunk 4]
```diff
@@ -48,6 +62,8 @@ hub_con %>%
 
 Note however that not all dplyr filtering options are available for all data types yet.
 
+You can see how in the above output the required `model_id`, `output_type`, `output_type_id` and `value` column names are all present, as is required for a `model_out_tbl` object. However, the output of the above expression, while it conforms to the `model_out_tbl` conventions, is actually returned just as a `tbl_df` or `tibble` object.
```
Comment (annakrystalli) [inline vignettes/articles/connect_hub.Rmd:65]:
```suggestion
You can see how in the above output the required `model_id`, `output_type`, `output_type_id` and `value` column names are all present, as is required for a `model_out_tbl` object. However, the output of the above expression, while conforming to `model_out_tbl` convention, is actually returned just as a `tbl_df` or `tibble` object.
```

**Review by annakrystalli — APPROVED** [review]
Overall looks good! Just made some minor suggestions. Feel free to ignore and merge in when ready.

File: vignettes/articles/connect_hub.Rmd (hunk 2 above)
Comment (nickreich) [reply vignettes/articles/connect_hub.Rmd:18]: That is true in some cases, but aren't there other cases where the `model_id` is just a single field identifier? I have lost track of where we ended up on this conversation.

File: vignettes/articles/connect_hub.Rmd (hunk 1 above)
Comment (nickreich) [reply vignettes/articles/connect_hub.Rmd:16]: ok. good point. I'll address this with an update.

**Pull request merged**